geonode/layers/views.py

```diff
@@ -379,7 +380,7 @@ def layer_upload(request, template='upload/layer_upload.html'):
         for _k in _keys:
             if _k in out:
                 if isinstance(out[_k], string_types):
-                    out[_k] = out[_k].encode(layer_charset, 'surrogateescape').decode('utf-8', 'surrogateescape')
+                    out[_k] = surrogate_escape_string(out[_k], layer_charset)                
```

Comment:
@abyssal there are trailing spaces at the end of the line. This won't make `flake8 geonode` pass the syntax checking tests. Can you please fix this?